lib/stoplight/data_store/memory.rb

```diff
@@ -22,7 +22,10 @@ def names
       end
 
       def get_all(config)
-        synchronize { [query_failures(config), @states[config.name]] }
+        [
+          synchronize { query_failures(config) },
+          synchronize { @states[config.name] }
+        ]
```

Comment:
This are two completely separated operations, no need to have synchronized access to get them both